solvebio/resource/object.py

```diff
@@ -406,20 +406,39 @@ def _object_list_helper(self, **params):
         items = self.all(client=self._client, **params)
         return items
 
-    def files(self, **params):
+    def files(self, query='', **params):
+        if query:
+            params['query'] = query
+        params.pop('object_type', None)
         return self._object_list_helper(object_type='file', **params)
 
-    def folders(self, **params):
+    def folders(self, query='', **params):
+        if query:
+            params['query'] = query
+        params.pop('object_type', None)
         return self._object_list_helper(object_type='folder', **params)
 
-    def datasets(self, **params):
+    def datasets(self, query='', **params):
+        if query:
+            params['query'] = query
+        params.pop('object_type', None)
         return self._object_list_helper(object_type='dataset', **params)
 
-    def objects(self, **params):
+    def objects(self, query='', **params):
+        if query:
+            params['query'] = query
         return self._object_list_helper(**params)
 
-    def ls(self, **params):
-        return self.objects(**params)
+    def ls(self, query='', **params):
+        query = query or params.pop('query', '')
+        return self.objects(query=query, **params)
+
+    def search(self, query='', **params):
+        query = query or params.pop('query', '')
+        # TODO
+        params['ancestor_id'] = self.id
+        params['limit'] = 1000
+        return super(Object, self).search(query, **params)
```

Comment:
I believe I left this TODO in here in order to determine if there should be a recursive option. Should the user be able to get anything in this folder, or just depth=0 objects.

 and/or `ancestor_id` needs to be replaced by `parent_id` (although the server may handle this)

This PR needs tests